doc/Programming-guide/Using-Kaa-endpoint-SDKs/C/SDK-ESP8266/index.md

```diff
@@ -5,4 +5,383 @@ permalink: /:path/
+
+In scope of this example, `main_task()` just calls `main()` function:
+
+~~~c
+    static void main_task(void *pvParameters)
+    {
+        (void)pvParameters;
+        main();
+        for(;;);
+    }
+~~~
+
+The `main()` function is defined in `src/kaa_demo.c` and starts minimal Kaa client using Kaa C SDK:
+
+~~~c
+    #include <stddef.h>
+    #include <kaa/kaa_error.h>
+    #include <kaa/kaa_context.h>
+    #include <kaa/platform/kaa_client.h>
+
+    static void loop_fn(void *context)
+    {
+        for(int i = 0;i<5;i++) {
+            printf("Hello, Kaa!\r\n");
+        }
+
+        kaa_client_stop(context);
+    }
+
+    int main(void)
+    {
+        printf("Initializing Kaa client\r\n");
+
+        kaa_client_t *kaa_client;
+
+        kaa_error_t error_code = kaa_client_create(&kaa_client, NULL);
+        if(error_code) {
+            printf("Failed to create Kaa client\r\n");
+            return 1;
+        }
+
+        error_code = kaa_client_start(kaa_client, loop_fn, (void*)kaa_client, 0);
+        if(error_code) {
+            printf("Failed to start Kaa main loop\r\n");
+            return 1;
+        }
+
+        kaa_client_destroy(kaa_client);
+        return 0;
+    }
+~~~
+
+### Build system overview
+
+The Kaa C SDK makes use of CMake build system and and though you can choose any build system of youir preference, it is recommended to use CMake for Kaa applications as well.
+This will make possible to tightly integrate your application's bulid system with Kaa SDK and use already provided ESP8266 toolchain file.
+
+#### CMakeLists.txt
+
+First, lets add Kaa SDK subdirectory, so that we can build it together with application.
+
+~~~CMake
+cmake_minimum_required(VERSION 3.0.2)
+
+project(kaa_demo C)
+
+# Add Kaa SDK directory
+add_subdirectory(libs/kaa)
+~~~
+
+Next, let's create a static library with demo source files.
+
+~~~CMake
+# Add source files
+add_library(kaa_demo_s STATIC user/user_main.c driver/uart.c src/kaa_demo.c)
+~~~
+
+We also enable c99 standard:
+
+~~~CMake
+set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -std=c99")
+~~~
+
+This part adds required include directories to compiler search path.
+
+~~~CMake
+if(NOT DEFINED ESP_RTOS_SDK)
+    set(ESP_RTOS_SDK /opt/Espressif/esp-rtos-sdk)
+endif()
+
+# specify include directories
+target_include_directories(kaa_demo_s PUBLIC driver)
+target_include_directories(kaa_demo_s PUBLIC .)
+target_include_directories(kaa_demo_s PUBLIC
+                           ${ESP_RTOS_SDK}/extra_include
+                           ${ESP_RTOS_SDK}/include
+                           ${ESP_RTOS_SDK}/include/lwip
+                           ${ESP_RTOS_SDK}/include/lwip/ipv4
+                           ${ESP_RTOS_SDK}/include/lwip/ipv6
+                           ${ESP_RTOS_SDK}/include/espressif/
+                           )
+~~~
+
+We also need to use custom linked script.
+Below, two linker scripts, required for ESP8266 applications, are moved to binary directory so that linker can locate them.
```

Comment:
That's not the purpose of these lines. They are used to configure linker scripts (i.e., replace CMake variable names with their values).